### PR TITLE
fix index out of range taking &b[0]

### PR DESCRIPTION
**Pull request by joesis**
Fix https://github.com/getlantern/http-proxy-lantern/issues/181, which was a silly bug I brought in at https://github.com/getlantern/tcpinfo/pull/1.

### CODE REVIEW

File: sys_linux.go [hunk 1]
```diff
@@ -142,11 +142,11 @@ func parseInfo(b []byte) (tcpopt.Option, error) {
 }
 
 func parseCCAlgorithmInfo(name string, b []byte) (CCAlgorithmInfo, error) {
-	pt := unsafe.Pointer(&b[0])
```
Comment (joesis) [inline sys_linux.go]: It panics when `b` is `nil`

**Comment by joesis** [comment]
@oxtoacart I'm merging directly as the change is trivial and fixes the panic in both `http-proxy-lantern` and `flashlight`, it doesn't trigger data race either.

**Comment by oxtoacart** [comment]
Thanks @joesis. If I'm correct, this only happens when running on a system that doesn't have BBR congestion control, does that sound right?

**Comment by joesis** [comment]
Not quite sure how the panic was triggered @oxtoacart . The method also cares about TCPVegas etc but our code seems call with `bbr` only.

**Pull request merged**